src/patterns/ProgressWizard/hooks/usePendingValue.tsx

```diff
@@ -0,0 +1,65 @@
+import { useState, useRef, useEffect, type Dispatch, type SetStateAction } from 'react';
+import { type PathValue, useFormContext, type FieldValues, type Path } from 'react-hook-form';
+
+/**
+ * Return type for the usePendingValue hook.
+ * @template T - The form shape as key value pairs
+ * @property setPendingValue - Setter for the pending value (call this to stage a value)
+ * @property applyPendingValue - Call this to commit the pending value to the form field
+ * @property pendingValue - The currently staged value (or undefined if none)
+ *
+ * @example
+ * const { setPendingValue, applyPendingValue, pendingValue } = usePendingValue('country');
+ */
+export type UsePendingValueReturn<T extends FieldValues> = {
+  setPendingValue: Dispatch<SetStateAction<PathValue<T, Path<T>> | undefined>>;
```

Comment:
I find this really hard to read. I am not the greatest typescripter but I wonder if we can't do anything to make this more readable. Even if it is just moving these types around a bit. 
```js
type FieldValue<T> = PathValue<T, Path<T>>;

export type UsePendingValueReturn<T extends FieldValues> = {
  setPendingValue: Dispatch<SetStateAction<FieldValue<T> | undefined>>;
  applyPendingValue: () => void;
  pendingValue: FieldValue<T> | undefined;
};
```